components/__init__.py

```diff
@@ -342,11 +342,13 @@ def define_components(reqs):
     retriever = GitRepoRetriever("https://github.com/spdk/spdk.git", True)
     reqs.define('spdk',
                 retriever=retriever,
-                commands=['./configure --prefix="$SPDK_PREFIX" ' \
-                          ' --with-fio="$FIO_SRC"',
+                commands=['./configure --prefix="$SPDK_PREFIX" --with-shared',
+                          #' --with-fio="$FIO_SRC" ' TODO: re-enable fio_plugin
                           'make $JOBS_OPT', 'make install',
+                          'cp -r include/spdk_internal "$SPDK_PREFIX/include"',
```

Comment:
This isn't necessary anymore, I'm going to reconstruct the BIO code to avoid using the SPDK internal API.

Reply:
This hack can be removed now, the spdk_set_thread() has been exported in the latest SPDK.